src/main/java/seedu/address/logic/commands/DeleteTransactionCommand.java

```diff
@@ -0,0 +1,70 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Transaction;
+
+/**
+ * Deletes a transaction from a person's transaction list using its specified index.
+ */
+public class DeleteTransactionCommand extends Command {
+    public static final String COMMAND_WORD = "deletet";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Deletes the transaction identified by the index number used in the displayed transaction list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_DELETE_TRANSACTION_SUCCESS = "Deleted transaction %1$s\\nfrom %2$s";
+
+    private static final Index CURRENT_PERSON = Index.fromOneBased(1);
+    private final Index index;
+
+
+    public DeleteTransactionCommand(Index index) {
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (CURRENT_PERSON.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person selected = lastShownList.get(CURRENT_PERSON.getZeroBased());
+        List<Transaction> transactions = selected.getTransactions();
+
+        if (index.getZeroBased() >= transactions.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TRANSACTION_DISPLAYED_INDEX);
+        }
+
+        Transaction transactionToRemove = transactions.get(index.getZeroBased());
+        transactions.remove(index.getZeroBased());
+
+        return new CommandResult(String.format(MESSAGE_DELETE_TRANSACTION_SUCCESS, transactionToRemove.toString(),
+                Messages.format(selected)));
```

Comment:
Can consider using `Message.format(transactionToRemove)` instead of `transactionToRemove.toString()` to standardise the message format
Currently the message for add transaction is this:
![image](https://github.com/user-attachments/assets/4ab12f1a-6b79-4087-b2f6-0e4ed8de3a98)

And the message for delete transaction is this:
![image](https://github.com/user-attachments/assets/f09a7793-6b34-4b58-9f16-22792eb3823a)